core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedGlobalAggregationWithProjection.java

```diff
@@ -175,16 +180,21 @@ public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Co
         }
 
         source = decorrelatedSource.get().getNode();
+        Optional<Symbol> nonNull = Optional.empty();
 
-        // append non-null symbol on nested plan. It will be used to restore semantics of null-sensitive aggregations after LEFT join
-        Symbol nonNull = context.getSymbolAllocator().newSymbol("non_null", BOOLEAN);
-        source = new ProjectNode(
-                context.getIdAllocator().getNextId(),
-                source,
-                Assignments.builder()
-                        .putIdentities(source.getOutputSymbols())
-                        .put(nonNull, TRUE)
-                        .build());
+        AggregationNode globalAggregation = captures.get(AGGREGATION);
+        // bool_or aggregation is not null-sensitive so no need to add additional mask
```

Comment:
nit: comment: `null` insensitive aggragrations don't distniquish empty input from null input, therefore they don't need a special mask `true` symbol to distinquish between these two cases after decorrelation. For example `count(*)` requires mask symbol becuase it returns 0 for empty set, but 1 for set with single null row., but `boolean_or` returns `null` for both empty set and set with single null row